### PR TITLE
std.math: use most inline x86/x64 assembly only if real is defined as x87

### PR DESCRIPTION
**Pull request by kinke**
Now that real == double on Win64. Useful in combination with https://github.com/ldc-developers/druntime/pull/13 and MSVC x64 2013+.

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -83,8 +83,32 @@ version(LDC)
...
+    version(LDC_LLVM_304)
+    {
+        version = INTRINSICS_FROM_303;
+        version = INTRINSICS_FROM_304;
+    }
+    version(LDC_LLVM_305)
+    {
+        version = INTRINSICS_FROM_303;
+        version = INTRINSICS_FROM_304;
+    }
```
Comment (redstar) [inline std/math.d:106]: `version = INTRINSICS_FROM_305;` is missing here.

File: std/math.d [hunk 2]
```diff
@@ -83,8 +83,32 @@ version(LDC)
...
+    version(LDC_LLVM_306)
+    {
+        version = INTRINSICS_FROM_303;
+        version = INTRINSICS_FROM_304;
+    }
```
Comment (redstar) [inline std/math.d:111]: Same here and `version = INTRINSICS_FROM_306;`!

**Comment by kinke** [comment]
No intrinsics introduced in LLVM 3.5/3.6 are used, that's why I limited the INTRINSICS_FROM_\* to those which are actually required.

**Comment by redstar** [comment]
Does not build on Linux:

```
/home/travis/build/ldc-developers/ldc/runtime/phobos/std/math.d(90): Error: version INLINE_YL2X defined after use
```

See https://travis-ci.org/ldc-developers/ldc/jobs/38571952

**Comment by kinke** [comment]
Hmm, no idea what's causing this error. I've simply moved the `version = INLINE_YL2X;` statements into a `static if (real.sizeof > double.sizeof) { ... }` block.
Additionally, the problem seems to arise while compiling `std.mathspecial`, which publicly imports `std.math`. And according to Travis output, that phobos `std.mathspecial` seems to be compiled while building druntime (?!), specifically, `rt.sections_linux` and/or `rt.sections_android`...

**Comment by redstar** [comment]
The Travis output is not ordered because several processes are running in parallel.

**Comment by redstar** [comment]
I think the problem here is that the `version` blocks are evaluated first. This is done because the code of a not satisfied version is not compiled into the program. In contrast, the `static if` is always evaluated at compile time. The result is that `version = INLINE_YL2X;` is evaluated after all version guards have been evaluated. The TDPL rule is: You can set a version only before any read of that version.

It works in the Win64 case because the body of the `static if` is excluded.

**Comment by kinke** [comment]
Should be fixed in https://github.com/ldc-developers/phobos/pull/4.

**Comment by dnadlinger** [comment]
I don't think duplicating the `INTRINSICS_FROM_…` logic here is a good idea. Instead, use `__traits(compiles, <… call to the intrinsic in question…>)`.

**Comment by redstar** [comment]
Good hint.

**Pull request merged**